### PR TITLE
[WIP] Add ansatz using tapered excitations to qubit tapering demo

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
This PR modifies the circuit ansatz in the tapering demo to now use tapered excitations. This update correction does not have any effect on the conclusion/results in the demo.

### CODE REVIEW

File: demonstrations/tutorial_qubit_tapering.py [hunk 1]
```diff
@@ -227,30 +227,41 @@ def circuit():
...
 dev = qml.device('default.qubit', wires=H_tapered.wires)
 @qml.qnode(dev)
 def circuit(params):
     qml.BasisState(state_tapered, wires=H_tapered.wires)
-    qml.PauliRot(params[2], 'Y',  wires=[0])
-    qml.PauliRot(params[1], 'Y',  wires=[1])
-    qml.PauliRot(params[0], 'YX', wires=[0, 1])
+    for idx, excitation in enumerate(tapered_excitations):
+        for itx, op in enumerate(excitation.ops):
+            qml.PauliRot(params[idx][itx],
```
Comment (soranjh) [inline demonstrations/tutorial_qubit_tapering.py:251]: The tapered excitations cannot be directly applied?

**Review by soranjh — CHANGES_REQUESTED** [review]
Thanks @obliviateandsurrender, left few comments. The main one is about the format of the tapered excitations. I think it is better to have tapered excitation gates that are directly applicable to the state not with `qml.PauliRot`. You might also consider this:

> If your demo uses the latest release of PennyLane, simply make your PR against the master branch. If you instead require the cutting-edge development versions of PennyLane or any relevant plugins, make your PR against the dev branch instead.

File: demonstrations/tutorial_qubit_tapering.py (hunk 1 above)
Comment (obliviateandsurrender) [reply demonstrations/tutorial_qubit_tapering.py]: It can be now!

**Comment by antalszava** [comment]
PR was closed automatically as the `dev` branch was recreated. Re-opened the PR and now it targets `master`.

If the demo does use any post-v0.26.0 features from PennyLane, the target branch should be changed again to `dev`.

File: demonstrations/tutorial_qubit_tapering.py [hunk 2]
```diff
@@ -227,30 +227,36 @@ def circuit():
...
     qml.BasisState(state_tapered, wires=H_tapered.wires)
-    qml.PauliRot(params[2], 'Y',  wires=[0])
-    qml.PauliRot(params[1], 'Y',  wires=[1])
-    qml.PauliRot(params[0], 'YX', wires=[0, 1])
+    for idx, double in enumerate(doubles):
+        qml.taper_operation(qml.DoubleExcitation(params[idx], wires=double),
+                                generators, paulixops, paulix_sector, H.wires)
+    for idx, single in enumerate(singles):
+        qml.taper_operation(qml.SingleExcitation(params[len(doubles) + idx], wires=single),
+                                generators, paulixops, paulix_sector, H.wires)
```
Comment (soranjh) [inline demonstrations/tutorial_qubit_tapering.py:247]: Does it need to be updated regarding this [PR](https://github.com/PennyLaneAI/pennylane/pull/3121)?

**Pull request closed without merging**